ckan/logic/validators.py

```diff
@@ -995,7 +995,7 @@ def email_validator(value: Any, context: Context) -> Any:
     if value:
         if not email_pattern.match(value):
             raise Invalid(_('Email {email} is not a valid format').format(email=value))
-    return value
+    return value.lower()
```

Comment:
@TomeCirun last time this was discussed we agreed to not change the user input (eg, if users input `ABC@example.com` we don't want to store `abc@example.com` in the database). Your changes here will force lower case in all incoming emails.

I don't know the best alternative off the top of my head, I don't have enough context as to why we dropped the `email_is_unique` validator in https://github.com/ckan/ckan/pull/6916 in favour of a db-based check but we need to give this further thought

Reply:
Hey @amercader ,

I recall why we removed the `email_is_unique` validator in **#6916**. The concern was that concurrent `user_update` or `user_create` requests with the same email might lead to duplicated entries.

As highlighted by Ian in the original [issue](https://github.com/ckan/ckan/issues/6893#issuecomment-1148684210), we opted to ensure unique active emails by implementing a composite constraint at the database level (which I implemented in the PR). We believed this database-level constraint alone would be sufficient, leading to the removal of the `email_is_unique` validator.

However, there's no harm in reintroducing `email_is_unique`. I've made some minor adjustments to address our requirements.

please let me know your opinion on this